Revert https://github.com/nodejs/node/commit/938212f3e74a74d0b436941aa24e71425ff666c5

Seems like host GCC compiler doesn't support this yet.
--- ./configure.py.orig	2022-10-06 06:09:46.883008809 +0530
+++ ./configure.py	2022-10-06 06:10:31.483008792 +0530
@@ -1245,10 +1245,6 @@
 
   o['variables']['want_separate_host_toolset'] = int(cross_compiling)
 
-  # Enable branch protection for arm64
-  if target_arch == 'arm64':
-    o['cflags']+=['-msign-return-address=all']
-
   if options.node_snapshot_main is not None:
     if options.shared:
       # This should be possible to fix, but we will need to refactor the
